src/mahoji/commands/steal.ts

```diff
@@ -0,0 +1,177 @@
+import { OSBMahojiCommand } from '../lib/util';
+import { getSkillsOfMahojiUser, mahojiUsersSettingsFetch } from '../mahojiSettings';
+
+export const stealCommand: OSBMahojiCommand = {
+	name: 'steal',
+	description: 'Sends your minion to steal to train Thieving.',
+	attributes: {
+		requiresMinion: true,
+		requiresMinionNotBusy: true,
+		examples: ['/steal name:Man']
+	},
+	options: [
+		{
+			type: ApplicationCommandOptionType.String,
+			name: 'name',
+			description: 'The object you try to steal from.',
+			required: true,
+			autocomplete: async (value: string, user: APIUser) => {
+				const mUser = await mahojiUsersSettingsFetch(user.id);
+				const conLevel = getSkillsOfMahojiUser(mUser, true).thieving;
+				return stealables
+					.filter(i => (!value ? true : i.name.toLowerCase().includes(value.toLowerCase())))
+					.filter(c => c.level <= conLevel)
+					.map(i => ({
+						name: i.name,
+						value: i.name
+					}));
+			}
+		},
+		{
+			type: ApplicationCommandOptionType.Integer,
+			name: 'quantity',
+			description: 'The quantity (defaults to max).',
+			required: false,
+			min_value: 1
+		}
+	],
+	run: async ({ options, userID, channelID }: CommandRunOptions<{ name: string; quantity?: number }>) => {
+		const user = await globalClient.fetchUser(userID);
+		stealables;
+		const stealable: Stealable | undefined = stealables.find(
+			obj =>
+				stringMatches(obj.name, options.name) || obj.aliases?.some(alias => stringMatches(alias, options.name))
+		);
+
+		if (!stealable) {
+			return `That is not a valid NPC/Stall to pickpocket or steal from, try pickpocketing or stealing from one of the following: ${stealables
+				.map(obj => obj.name)
+				.join(', ')}.`;
+		}
+
+		if (stealable.qpRequired && user.settings.get(UserSettings.QP) < stealable.qpRequired) {
+			return `You need atleast **${stealable.qpRequired}** QP to ${
+				stealable.type === 'pickpockable' ? 'pickpocket' : 'steal from'
+			} a ${stealable.name}.`;
+		}
+
+		if (stealable.itemsRequired && !user.allItemsOwned().has(stealable.itemsRequired)) {
+			return `You need these items to ${
+				stealable.type === 'pickpockable' ? 'pickpocket this NPC' : 'steal from this stall'
+			}: ${new Bank(stealable.itemsRequired)}.`;
+		}
+
+		if (user.skillLevel(SkillsEnum.Thieving) < stealable.level) {
+			return `${user.minionName} needs ${stealable.level} Thieving to ${
+				stealable.type === 'pickpockable' ? 'pickpocket' : 'steal from'
+			} a ${stealable.name}.`;
+		}
+
+		const [hasFavour, requiredPoints] = gotFavour(user, Favours.Hosidius, 15);
+		if (!hasFavour && stealable.name === 'Fruit stall') {
+			return `${user.minionName} needs ${requiredPoints}% Hosidius Favour to steal fruit from the Fruit stalls!`;
+		}
+
+		const timeToTheft =
+			stealable.type === 'pickpockable' ? (stealable.customTickRate ?? 2) * 600 : stealable.respawnTime;
+
+		if (!timeToTheft) {
+			return 'This NPC/Stall is not working as intended, report it.';
```

Comment:
dont do this - use `logError` so we can actually know if something like this happens